daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -155,7 +164,9 @@ object Node {
       exerciseResult: Option[Value],
       key: Option[KeyWithMaintainers],
       override val byKey: Boolean, // invariant (!byKey || exerciseResult.isDefined)
-      // For the sake of consistency between types with a version field, keep this field the last.
+                                   // For the sake of consistency between types with a version field, keep this field the last.
+      byImplementation: Option[Implementation], // The implementation of the interface choice if any.
```

Comment:
Is there the invariant "`byImplementation.isDefined` implies `interfaceId.isDefined`?

Reply:
yes

Reply:
Great. then it would be good to enforce this invariant in this class with a `require` or `assert` or similar.

Reply:
added a require